extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -15,26 +15,78 @@
  */
 
 import {CSS} from '../../../build/amp-story-panning-media-0.1.css';
+import {CommonSignals} from '../../../src/common-signals';
 import {Layout} from '../../../src/layout';
+import {dev, user} from '../../../src/log';
+import {setStyles} from '../../../src/style';
+import {whenUpgradedToCustomElement} from '../../../src/dom';
+
+/** @const {string} */
+const TAG = 'AMP_STORY_PANNING_MEDIA';
 
 export class AmpStoryPanningMedia extends AMP.BaseElement {
   /** @param {!AmpElement} element */
   constructor(element) {
     super(element);
 
-    /** @private {string} */
-    this.myText_ = 'hello world';
+    /** @private {?Element} */
+    this.element_ = element;
 
     /** @private {?Element} */
     this.container_ = null;
+
+    /** @private {?Element} */
+    this.image_ = null;
+
+    /** @private {?string} */
+    this.x_ = element.getAttribute('x') || '0%';
+
+    /** @private {?string} */
+    this.y_ = element.getAttribute('y') || '0%';
+
+    /** @private {?string} */
+    this.zoom_ = element.getAttribute('zoom') || '1';
   }
 
   /** @override */
-  buildCallback() {
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    this.applyFillContent(this.container_, /* replacedContent */ true);
+  buildCallback() {}
+
+  /** @override */
+  layoutCallback() {
+    const ampImgEl = this.element_.querySelector('amp-img');
+    return whenUpgradedToCustomElement(ampImgEl)
+      .then(() => {
+        return ampImgEl.signals().whenSignal(CommonSignals.LOAD_END);
+      })
```

Comment:
You can remove the `{}`and `return` here and use `() => ampImgEl.signals().whenSignal...`